.github/workflows/terraform.yaml

```
@@ -0,0 +1,54 @@
+name: terraform
+
+on:
+  - pull_request
+
+jobs:
+  terraform:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install terraform
+      uses: hashicorp/setup-terraform@v1
+
+    - name: Terraform Init
+      id: init
+      run: terraform init
+      working-directory: ./test
+
+    - name: Terraform Plan
+      id: plan
+      run: terraform plan -no-color
+      continue-on-error: true
+      working-directory: ./test
+
+    - uses: actions/github-script@0.9.0
+      if: github.event_name == 'pull_request'
+      env:
+        PLAN: "terraform\n${{ steps.plan.outputs.stdout }}"
+      with:
+        github-token: ${{ secrets.GITHUB_TOKEN }}
+        script: |
+          const output = `#### Terraform Format and Style\
+          🖌\`${{ steps.fmt.outcome }}\`
+          #### Terraform Initialization
+          ⚙️\`${{ steps.init.outcome }}\`
+          #### Terraform Validation
+          🤖${{ steps.validate.outputs.stdout }}
+          #### Terraform Plan
+          📖\`${{ steps.plan.outcome }}\`
```

```suggestion
          const output = `- [${{steps.init.outputs.exitcode == 0 ? 'X' : ''}}] Terraform Initialization
          - [${{steps.validate.outputs.exitcode == 0 ? 'X' : ''}}] Terraform Validation
          - [${{steps.plan.outputs.exitcode == 0 ? 'X' : ''}}] Terraform Plan
```

---

I haven't verified this at all, but let's see if this would work. the quotes might be a bit off, but this should show a checkbox when things worked out right. We could also swap that for checks on the output like so:

```suggestion
          const output = `${{steps.init.outputs.exitcode == 0 ? ':x:' : ':white_check_mark:'}} Terraform Initialization
          ${{steps.validate.outputs.exitcode == 0 ? ':x:' : ':white_check_mark:'}} Terraform Validation
          ${{steps.plan.outputs.exitcode == 0 ? ':x:' : ':white_check_mark:'}} Terraform Plan
```